File: backend/FitHub/src/main/java/com/uche/fithub/dto/subscription_dto/SubscriptionDto.java
package com.uche.fithub.dto.subscription_dto;

import java.time.LocalDate;

import com.uche.fithub.dto.customer_dto.CustomerDto;
import com.uche.fithub.dto.pack_dto.PackDto;
import lombok.Data;

@Data
public class SubscriptionDto {
    private Long id;

    private CustomerDto customer;

    private PackDto pack;

    private LocalDate startDate;
    private boolean active;
}

File: frontend/fithub_front/src/app/components/subscription/models/SubscriptionDto.ts
import { PackResponseSchema } from '../../pack/model/PackResponseSchema';
import { CustomerDto } from './../../customer/models/Customer';

export class SubscriptionDto {
  id: number;
  customer: CustomerDto;
  pack: PackResponseSchema;
  startDate = new Date();
  endDate = new Date();
  active: boolean;

  constructor(
    id: number = 0,
    customer: CustomerDto,
    startDate: Date = new Date(),
    endDate: Date = new Date(),
    active: boolean = false,
    pack: PackResponseSchema
  ) {
    this.id = id;
    this.customer = customer;
    this.startDate = startDate;
    this.active = active;
    this.pack = pack;
    this.endDate = endDate;
  }
}
